### PR TITLE
Mrb/add telemetry and timeout tests

### PR DESCRIPTION
**Pull request by mikowitz**
Add tests for telemetry endpoints and for handling timeouts correctly.

The first commit is simply creating the `tests/0.2.2` directory, the [second](08ba4954bc4082d7db68d04adda8d2ecc0862a70) has the new test data.

### CODE REVIEW

File: tests/0.2.2/post.yaml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        collect_sync_interval: 1000
+
+    - name: reports example contexts
+      client: client
+      function: post
+      aggregator: example_contexts
+      data:
+        - user:
+            name: "michael"
+            age: 38,
```
Comment (semanticart) [inline tests/0.2.2/post.yaml:110]: This `,` typo should be removed. Ruby's YAML implementation treats it as `38` but Node's YAML implementation treats it as the string `"38,"`

File: tests/0.2.2/post.yaml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        - logger_name: "my.path"
+          counts:
+            debugs: 3
+            infos: 1
+        - logger_name: "another.path"
+          counts:
+            warns: 1
+            errors: 2
+      client_overrides:
+        collect_sync_interval: 600
```
[the remark this answers is not in the document]
Comment (mikowitz) [reply tests/0.2.2/post.yaml]: apparently not...

File: tests/0.2.2/post.yaml [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            config_row_index: 0
+            conditional_value_index: 1
+        - key: "feature-flag.weighted"
+          type: FEATURE_FLAG
+          value: 2
+          value_type: int
+          count: 1
+          summary:
+            config_row_index: 0
+            weighted_value_index: 2
```
Comment (semanticart) [inline tests/0.2.2/post.yaml:108]: Can we add `conditional_value_index: 0` here for consistency?

**Review by semanticart — APPROVED** [review]
Thank you

**Pull request merged**